controllers/storagecluster/provider_server.go

```diff
@@ -496,9 +496,12 @@ func (o *ocsProviderServer) createJob(r *StorageClusterReconciler, instance *ocs
 
 	if errors.IsNotFound(err) {
 		onboardingSecretGeneratorJob := getOnboardingJobObject(instance)
-		err = r.Client.Create(context.Background(), onboardingSecretGeneratorJob)
-	}
-	if err != nil {
+		jobErr := r.Client.Create(context.Background(), onboardingSecretGeneratorJob)
+		if jobErr != nil && !errors.IsAlreadyExists(jobErr) {
```

Comment:
I believe you might have hit this during very quick reconciles that matches the length of job duration and I agree we don't want to keep on queuing reconciles and at the same time we are unsure whether the job status is completed due to failure or success.

So, I'd like to settle for a middle ground if you will, send a reconcile with delay if job already exists? then the issue will be, who'll ensure updating this if the job responsibilities increases and I'd argue that's better than not getting a reconcile at all.

At the end, everything needs to stabilize unless the garbage collector controller give up and not removing pods, basically not respecting ttl (happens more often than we think) and storagecluster will not progress.

Sry for being a li'l exhaustive and if every solution has edge cases I believe we want to take action once we get some regression results from testing and settle on fixing the case w/ most probability, wdyt?

Reply:
Are you proposing a requeue? If so then yeah that makes sense.

Reply:
Sry, by the end I meant, there isn't any fix known yet and we wait for regression tests to see what edge case is more frequent and fix that.

Reply:
@leelavg @jarrpa Please take a look at this comment: https://github.com/red-hat-storage/ocs-operator/pull/2500#discussion_r1521399363

We already have Ritesh looking at the problem from a different angle

Reply:
ACK

Reply:
Done.